### PR TITLE
Allocate hexagon runtime arguments buffers on the heap if there are too many arguments

### PR DESCRIPTION
**Pull request by white-pony**
I also included some of the binaries, I wasn't exactly sure which one of them are needed, so please let me know if I missed anything or something is not needed.

Also, I don't know if it matters, but Hexagon SDK I have comes with Android NDK r14b instead of r10d from Makefile, so I built with that one (with local modifications to Makefile).

### CODE REVIEW

File: src/runtime/hexagon_remote/halide_remote.cpp [hunk 1]
```diff
@@ -368,6 +385,11 @@ int halide_hexagon_remote_run_v2(handle_t module_ptr, handle_t function,
     // Power HVX off.
     halide_hexagon_remote_power_hvx_off();
 
+    if (allocated_on_heap) {
+        free(buffers);
```
Comment (steven-johnson) [inline src/runtime/hexagon_remote/halide_remote.cpp]: The `return` statement on line 379 will skip these frees.

**Comment by steven-johnson** [comment]
LGTM

**Comment by dsharletg** [comment]
Looks good, but libhalide_hexagon_host.so doesn't need to change.

**Comment by white-pony** [comment]
> Looks good, but libhalide_hexagon_host.so doesn't need to change.

Thanks, was not sure about this one, reverted it back.

**Pull request merged**